pkg/token/provider.go

```diff
@@ -24,13 +24,17 @@ func newTokenProvider(o *Options) (TokenProvider, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to get cloud.Configuration. err: %s", err)
 	}
+	popClaimsMap, err := parsePopClaims(o.PoPTokenClaims)
+	if o.IsPoPTokenEnabled && err != nil {
+		return nil, fmt.Errorf("failed to parse pop token claims. err: %w", err)
```

Comment:
the error msg might be a duplicate based on what `parsePopClaims` function will returns.

`failed to parse pop token claims. err: error parsing PoP token claims. Ensure the claims are formatted as `key=value` with no extra whitespace`

Reply:
Updated the error messages in `parsePopClaims` and removed the additional error wrapping here since those are already descriptive